src/collections/_documentation/development/sdk-dev/envelopes.md

```diff
@@ -0,0 +1,448 @@
+characters are literal.
+
+**Envelope with 2 items:**
+
+Note that the attachment contains a Windows newline at the end of its
+payload which is included in `length`:
+
+```
+{"event_id":"9ec79c33ec9942ab8353589fcb2e04dc","dsn":"https://e12d836b15bb49d7bbf99e64295d995b:@sentry.io/42"}\n
+{"type":"attachment","length":10,"content_type":"text/plain","filename":"hello.txt"}\n
+\xef\xbb\xbfHello\r\n\n
+{"type":"event","length":41,"content_type":"application/json","filename":"application.log"}\n
+{"message":"hello world","level":"error"}\n
+```
+
+**Empty Envelope without items, last newline omitted:**
+
+Note that the attachment contains a Windows newline at the end of its
+payload which is included in `length`:
+
+```
+{"event_id":"9ec79c33ec9942ab8353589fcb2e04dc","dsn":"https://e12d836b15bb49d7bbf99e64295d995b:@sentry.io/42"}\n
+{"type":"attachment","length":10,"content_type":"text/plain","filename":"hello.txt"}\n
+\xef\xbb\xbfHello\r\n\n
+{"type":"event","length":41,"content_type":"application/json","filename":"application.log"}\n
+{"message":"hello world","level":"error"}
+```
+
+**Envelope with 2 empty attachments:**
+
+```
+{"event_id":"9ec79c33ec9942ab8353589fcb2e04dc"}\n
+{"type":"attachment","length":0}\n
+\n
+{"type":"attachment","length":0}\n
+\n
+```
+
+**Envelope with 2 empty attachments, last newline omitted:**
+
+```
+{"event_id":"9ec79c33ec9942ab8353589fcb2e04dc"}\n
+{"type":"attachment","length":0}\n
+\n
+{"type":"attachment","length":0}\n
+```
+
+**Item with implicit length, terminated by newline:**
+
+```
+{"event_id":"9ec79c33ec9942ab8353589fcb2e04dc"}\n
+{"type":"attachment"}\n
+helloworld\n
+```
+
+**Item with implicit length, last newline omitted, terminated by EOF:**
+
+```
+{"event_id":"9ec79c33ec9942ab8353589fcb2e04dc"}\n
+{"type":"attachment"}\n
+helloworld
+```
+
+**Envelope without headers, implicit length, last newline omitted, terminated by
+EOF:**
+
+```
+{}\n
+{"type":"session"}\n
+{"started": "2020-02-07T14:16:00Z","attrs":{"release":"sentry-test@1.0.0"}}
+```
+
+## Data Model
+
+Each Envelope consists of headers and a potentially empty list of items, each
+with their own headers. Which Headers are required depends on the Items in an
+Envelope. This section describes all Item types and their respective required
+headers.
+
+The type of an item is declared in the `type` header, as well as the payload
+size in `length`. See Serialization Format for a list of common Item headers.
+The headers described in this section are **in addition to the common headers**.
+
+### Event
+
+Item type `"event"`. This Item contains an error or default event payload
+encoded in JSON.
+
+**Constraints:**
+
+- This item may occur at most once per Envelope.
+- This item is mutually exclusive with `"transaction"` items.
+
+**Envelope Headers:**
+
+`event_id`
+
+: **UUID String, required.** Corresponds to the `event_id` field of the event
+  payload. Clients are required to generate an event id ahead of time and set it
```

Comment:
because ID is an abbreviation, it should be capitalized (unless it's in a code sample that has issues with use of capitals).